src/components/shared/WithSize.js

```diff
@@ -45,7 +61,12 @@ export function withSize<
         throw new Error('Unable to find the DOMNode');
       }
       this._resizeListener = () => {
-        this._updateWidth(container);
+        debounce(
+          requestIdleCallbackPolyfill(() => {
+            this._updateWidth(container);
+          }, this._requestIdleTimeout),
```

Comment:
you don't use debounce correctly here. Debounce takes a function, but here you pass the _result_ of the call to `requestIdleCallbackPolyfill`. As a result `requestIdleCallbackPolyfill` is called at each resize and we don't use the effect of `debounce` at all.

This is how you can use it:
```js
debounce(
  () => requestIdleCallbackPolyfill(...),
  2000,
);
```

I hope this is clearer now!

Flow can help if you type the arguments of `debounce`.

Reply:
Thanks a lot! Now it is clear.
And I used `debounce` in the `EventListener` now.